### ISSUE TITLE
string_decoder dependency is being flagged as vulnerability

### ISSUE THREAD

**Issue by nathank527** [Azure/oav#1035]
Azure Dev Ops Component Governance is flagging our builds due to a vulnerability in the string_decoder dependency needed for OAV.  Is there any way this dependency can be replaced in favor of something that doesn't have said issue?  

Here is the Component Governance description: 

`This module is a mirror of the Node built-in StringDecoder object, and should only be used when operating on a very old version of Node or within a web- browser context. In addition, this object does not zero out memory before returning it, so it risks information disclosure.`

![image](https://github.com/Azure/oav/assets/31480019/382045d7-3acf-47da-8d8a-8c24908fb20e)

**Comment by scbedd** [comment]
![image](https://github.com/Azure/oav/assets/45376673/e8ad0a01-5f11-4783-8b29-0cb8e5532f35)

Unfortunately this isn't a straightforward fix. `copyfiles` can be fairly easily replaced. However, we use `winston` for all of our logging. That dependency is a bit nastier as it is used in a whole bunch of places, including as a dependent package within a server environment. (`winston` -> `readable-stream` -> `string_decoder`)

I'll be tackling this, but not a quick turnaround.

**Comment by nathank527** [comment]
Hi @scbedd, any update on this?

**Comment by scbedd** [comment]
> Hi [@scbedd](https://github.com/scbedd), any update on this?

Oi vey you're right. Apologies I got absolutely run over by Spring Grove response in June, and this slipped off my radar.

I'll add this to my action item list and see if I can get you an update in the next week or so.

**Comment by scbedd** [comment]
@nathank527 We're using `winston` for our logger, and it is not only a solid library, but is under active development. I don't think we're going to change that for a `low` CG warning.

If I'm reading the vulnerability correctly, `string_decoder` allocates a buffer, and it's possible that the buffer contains 4bytes of information for disclosure.

Further reading of the governance report states that

> If you are not using string_decoder directly, it is a sub-dependency, you can dismiss this alert

Not only that, but I went and found the usage of `string_decoder`, and it aligns exactly with a safe way to use it. Decoder is never accessed directly by external code, and a `buffer` object is iterated across to [power the decoding.](https://github.com/nodejs/readable-stream/blob/a2e9aedf4aeee4a5e4d8efcb175edb67e2817eaa/lib/internal/streams/readable.js#L346)

Not only that, but if you're over node 15 this isn't a problem either. `oav` is only used on node 16 and above.

[Component governance](https://dev.azure.com/azure-sdk/internal/_componentGovernance/216762/alert/11310208?typeId=25501023&pipelinesTrackingFilter=0) alert for this is `low`, and can be dismissed. I recommend you just dismiss. Closing as won't fix.
